web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java

```diff
@@ -13,4 +13,6 @@
 public interface IPostRepository extends JpaRepository<Post, Long>
 {
     List<Post> findAllByUserId(Long userId);
+    
+    void deleteAllByUserId(Long userId);

```

Comment:
```suggestion
    @Transactional
    void deleteAllByUserId(Long userId);
```

Spring JPA will not let you delete an item without this annotation to let the system know there is a transactional operation.